pkg/solana/chainwriter/transform_registry.go

```diff
@@ -90,3 +130,81 @@ func CCIPCommitAccountTransform(ctx context.Context, args any, accounts solana.A
 	}
 	return args, transformedAccounts, nil
 }
+
+func fetchPoolLookupAccounts(ctx context.Context, client client.MultiClient, poolTables map[string][]*solana.AccountMeta) ([]*solana.AccountMeta, error) {
+	var poolAccounts []*solana.AccountMeta
+	// poolTables only contains a single lookup table for token admin registry
+	for _, table := range poolTables {
+		tokenAdminRegistryPDA := table[1].PublicKey
+
+		// load token admin registry
+		resp, err := client.GetAccountInfoWithOpts(ctx, tokenAdminRegistryPDA, &rpc.GetAccountInfoOpts{
+			Encoding:   "base64",
+			Commitment: rpc.CommitmentFinalized,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch token admin registry account: %w", err)
+		}
+		tokenAdminRegistry := ccip_router.TokenAdminRegistry{}
+		err = bin.NewBorshDecoder(resp.GetBinary()).Decode(&tokenAdminRegistry)
+		if err != nil {
+			return nil, fmt.Errorf("failed to borsh decode token admin registry account: %w", err)
+		}
+
+		// lookup tables can store 256 addresses
+		// token admin registry's WritableIndexes field is the binary representation of indexes that are writable stored in 2 separate uint128 on-chain
+		writableBytes := append(tokenAdminRegistry.WritableIndexes[0].Bytes(), tokenAdminRegistry.WritableIndexes[1].Bytes()...)
```

Comment:
why are there two separate writable index bitmaps?

Reply:
Not sure why on-chain did it this way but they stored the writable bits in uint128's. Since lookup tables can only have 256 addresses, they have to use 2 [here](https://github.com/smartcontractkit/chainlink-ccip/blob/a8f5bd293fe5b6ecdd177124b29b198096ff2dc2/chains/solana/contracts/programs/ccip-offramp/src/instructions/v1/messages.rs#L82).

Reply:
it's to save on space since we only need one bit per account, rather than a whole byte